lib/livebook/secrets.ex

```diff
@@ -107,4 +118,9 @@ defmodule Livebook.Secrets do
   defp broadcast_secrets_change(message) do
     Phoenix.PubSub.broadcast(Livebook.PubSub, "secrets", message)
   end
+
+  defp to_struct(%{name: name, value: value} = fields) do
+    origin = fields[:origin] || :system_env
```

Comment:
I think the default here should be `:app`. After all, the `:system_env` ones never go to storage.